appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/EJBContainerTransactionManager.java

```diff
@@ -110,63 +100,63 @@
      * tx attributes can be looked up with variations of getTxAttr()
      */
     int findTxAttr(MethodDescriptor md) {
-        int txAttr = -1;
-
-        if ( container.isBeanManagedTran ) {
+        if (container.isBeanManagedTran) {
             return Container.TX_BEAN_MANAGED;
         }
-
-        ContainerTransaction ct = ejbDescriptor.getContainerTransactionFor(md);
-
-        if ( ct != null ) {
-            String attr = ct.getTransactionAttribute();
-            if ( attr.equals(ContainerTransaction.NOT_SUPPORTED) )
-                txAttr = Container.TX_NOT_SUPPORTED;
-            else if ( attr.equals(ContainerTransaction.SUPPORTS) )
-                txAttr = Container.TX_SUPPORTS;
-            else if ( attr.equals(ContainerTransaction.REQUIRED) )
-                txAttr = Container.TX_REQUIRED;
-            else if ( attr.equals(ContainerTransaction.REQUIRES_NEW) )
-                txAttr = Container.TX_REQUIRES_NEW;
-            else if ( attr.equals(ContainerTransaction.MANDATORY) )
-                txAttr = Container.TX_MANDATORY;
-            else if ( attr.equals(ContainerTransaction.NEVER) )
-                txAttr = Container.TX_NEVER;
+        final int txAttr = getTransactionAttribute(md);
+        if (txAttr < 0) {
+            throw new EJBException("Transaction Attribute not found for method " + md.prettyPrint());
         }
-
-        if ( txAttr == -1 ) {
-            throw new EJBException("Transaction Attribute not found for method "
-                + md.prettyPrint());
-        }
-
         container.validateTxAttr(md, txAttr);
-
         return txAttr;
     }
 
+    private int getTransactionAttribute(final MethodDescriptor md) {
+        final ContainerTransaction containerTx = ejbDescriptor.getContainerTransactionFor(md);
+        if (containerTx == null) {
+            return -1;
+        }
+        final String attr = containerTx.getTransactionAttribute();
+        if (attr.equals(ContainerTransaction.NOT_SUPPORTED)) {
+            return Container.TX_NOT_SUPPORTED;
+        } else if (attr.equals(ContainerTransaction.SUPPORTS)) {
+            return Container.TX_SUPPORTS;
+        } else if (attr.equals(ContainerTransaction.REQUIRED)) {
+            return Container.TX_REQUIRED;
+        } else if (attr.equals(ContainerTransaction.REQUIRES_NEW)) {
+            return Container.TX_REQUIRES_NEW;
+        } else if (attr.equals(ContainerTransaction.MANDATORY)) {
+            return Container.TX_MANDATORY;
+        } else if (attr.equals(ContainerTransaction.NEVER)) {
+            return Container.TX_NEVER;
+        } else {
+            return -2;
```

Comment:
It is only a dummy result, anything under a zero is an error and would cause an EJBException without this value. So it does not matter, maybe with debugger.

Reply:
In that case either extract a constant or put a comment explaining it - otherwise its just you knowing this.

Reply:
I will leave it as it is this time to spare some time of possible next iteration, because it is simple to find the only place where it is called and it is a private method not providing subzero results to other than nearest caller (which will throw an exception then).
But I agree with you :-)